R/todo_comment_linter.R

```diff
@@ -3,22 +3,24 @@
 #' Check that the source contains no TODO comments (case-insensitive).
 #'
 #' @param todo Vector of case-insensitive strings that identify TODO comments.
+#' @param except_regex Vector of case-sensitive regular expressions that identify
+#'    _valid_ TODO comments.
 #'
 #' @examples
 #' # will produce lints
 #' lint(
-#'   text = "x + y # TODO",
-#'   linters = todo_comment_linter()
+#'   text = "x + y # TOODOO",
+#'   linters = todo_comment_linter(todo = "toodoo")
 #' )
 #'
 #' lint(
-#'   text = "pi <- 1.0 # FIXME",
-#'   linters = todo_comment_linter()
+#'   text = "pi <- 1.0 # FIIXMEE",
+#'   linters = todo_comment_linter(todo = "fiixmee")
 #' )
 #'
 #' lint(
-#'   text = "x <- TRUE # hack",
-#'   linters = todo_comment_linter(todo = c("todo", "fixme", "hack"))
+#'   text = "x <- TRUE # TOODOO(#1234): Fix this hack.",
+#'   linters = todo_comment_linter()
```

Comment:
Why this?

Reply:
NB: diff is screwing this up a bit. This is just deleting an example. I think that example is pretty redundant now that the earlier two examples also introduce custom `todo`. Earlier, the first two examples relied on defaults and so this example was showing something new; now, not so I think.